File: framework/languages/dotnet/e2e/RegistryMessaging/Client/Scenarios/RmA2ManualEndpointScenario.cs
using Zlink.HttpClient;
using RegistryMessaging.Shared;
using RegistryMessaging.Client;

namespace RegistryMessaging.Client.Scenarios;

// RM-A2 verifies that a role server endpoint can request a profile provider through an
// explicitly configured channel endpoint without registry discovery.
internal static class RmA2ManualEndpointScenario
{
    public static async Task RunAsync(ZLinkHttpClient providerA)
    {
        var reply = (await providerA.Post("/profile/manual").Body(new ProfileRequest("rm-a2")).SubmitAsync<ProfileReply>()).Body;
        ScenarioAssert.That(reply.Value == "profile:rm-a2", "RM-A2 reply value mismatch.");
        ScenarioAssert.That(reply.ProviderRid == "api-a", "RM-A2 manual endpoint should reach api-a.");

        var evidence = (await providerA.Post("/evidence/wait")
            .Body(new EvidenceWaitRequest("value=rm-a2"))
            .SubmitAsync<string[]>()).Body;
        ScenarioAssert.That(
            evidence.Any(line => line.Contains("value=rm-a2", StringComparison.Ordinal)),
            "RM-A2 api-a evidence missing.");
        Console.WriteLine("scenario RM-A2 passed");
    }
}

File: framework/languages/node/e2e/RegistryMessaging/Client/Scenarios/rm-a2-manual-endpoint-scenario.ts
// RM-A2: 수동 endpoint 연결 (대조군) 시나리오를 검증한다.
import type { ProfileRes } from '../../Shared/messages';
import { postJson } from '../../../http-client';
import { ensure } from '../Support/scenario-assert';

export async function runRmA2(providerAUrl: string): Promise<void> {
  const reply = await postJson<ProfileRes>(providerAUrl, '/profile/manual', { value: 'rm-a2' });
  ensure(reply.value === 'profile:rm-a2', 'RM-A2 reply value mismatch.');
  ensure(reply.providerRid === 'api-a', 'RM-A2 manual endpoint should reach api-a.');
  const evidence = await postJson<string[]>(providerAUrl, '/evidence/wait', { contains: 'value=rm-a2' });
  ensure(evidence.some((line) => line.includes('value=rm-a2')), 'RM-A2 api-a evidence missing.');
  console.log('scenario RM-A2 passed');
}
